tests/kernel_bundle/sycl_is_compatible.cpp

```diff
@@ -0,0 +1,240 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+//  Provides tests for is_compatible()
+//
+*******************************************************************************/
+
+#include "kernel_bundle.h"
+#include "kernels.h"
+
+TEST_CASE(
+    "Check is_compatible(const std::vector<kernel_id>, const device) for zero "
+    "kernels",
+    "[kernel_bundle]") {
+  const sycl::device device = sycl_cts::util::get_cts_object::device();
+  const std::vector<sycl::kernel_id> kernelIds;
+
+  CHECK(sycl::is_compatible(kernelIds, device));
+}
+
+TEST_CASE(
+    "Check is_compatible(const std::vector<kernel_id>, const device) for "
+    "built-in kernel ids for the selected device",
+    "[kernel_bundle]") {
+  const sycl::device device = sycl_cts::util::get_cts_object::device();
+  const std::vector<sycl::kernel_id> builtinKernelIds =
+      device.get_info<sycl::info::device::built_in_kernel_ids>();
+
+  CHECK(sycl::is_compatible(builtinKernelIds, device));
+}
+
+TEST_CASE(
+    "Check is_compatible(const std::vector<kernel_id>, const device) for "
+    "built-in kernel ids from every root device other than the selected device",
+    "[kernel_bundle]") {
+  const sycl::device device = sycl_cts::util::get_cts_object::device();
+  auto platforms = sycl::platform::get_platforms();
+  std::vector<sycl::kernel_id> builtinKernelIds;
+
+  for (auto p : platforms) {
+    auto devices = p.get_devices();
+    for (auto d : devices) {
+      if (d != device) {
+        std::vector<sycl::kernel_id> newKernelIds =
+            d.get_info<sycl::info::device::built_in_kernel_ids>();
+        builtinKernelIds.insert(builtinKernelIds.end(), newKernelIds.begin(),
+                                newKernelIds.end());
+      }
+    }
+  }
+  if (!builtinKernelIds.empty())
+    CHECK(!sycl::is_compatible(builtinKernelIds, device));
```

Comment:
This seems to be a good case for SKIP macro.
```suggestion
  if (builtinKernelIds.empty())
    SKIP("No built-in kernels available for non-tested devices.");
  CHECK(!sycl::is_compatible(builtinKernelIds, device));
```

Reply:
Changed to SKIP.